src/simtools/applications/calculate_incident_angles.py

```diff
@@ -0,0 +1,194 @@
+        nargs="+",
+        required=False,
+    )
+    config.parser.add_argument(
+        "--source_distance",
+        help="Source distance in kilometers",
+        type=float,
+        default=10.0,
+        required=False,
+    )
+    config.parser.add_argument(
+        "--number_of_photons",
+        help="Number of star photons to trace (per run)",
+        type=int,
+        default=10000,
+        required=False,
+    )
+    config.parser.add_argument(
+        "--mirror_reflection_random_angle",
+        help="Set mirror_reflection_random_angle in degrees",
+        type=float,
+        required=False,
+    )
+    config.parser.add_argument(
+        "--mirror_alignment_random",
+        help="Accuracy of mirror alignment",
+        type=float,
+        default=0.005,
+        required=False,
+    )
+    config.parser.add_argument(
+        "--perfect_mirror",
+        help="Assume perfect mirror shape/alignment/reflection",
+        action="store_true",
+        required=False,
+    )
+    config.parser.add_argument(
+        "--debug_plots",
+        dest="debug_plots",
+        help="Generate additional debug plots (radius histograms, XY heatmaps, radius vs angle)",
+        action="store_true",
+        required=False,
+    )
+    config.parser.add_argument(
+        "--calculate_primary_secondary_angles",
+        dest="calculate_primary_secondary_angles",
+        help="Also compute angles of incidence on primary and secondary mirrors",
+        required=False,
+        action="store_true",
+    )
+    return config.initialize(
+        db_config=True,
+        simulation_model=["telescope", "site", "model_version"],
+    )
+
+
+def main():
+    """Application to calculate incident angles using ray tracing."""
+    label = Path(__file__).stem
+    args_dict, db_config = _parse(label)
+
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+
+    logger.info("Starting calculation of incident angles")
+
+    output_base = Path(args_dict.get("output_path", "./"))
+    output_dir = (
+        output_base / label if not args_dict.get("use_plain_output_path", False) else output_base
+    )
+    base_label = args_dict.get("label", label)
+    telescope_name = args_dict["telescope"]
+    label_with_telescope = f"{base_label}_{telescope_name}"
+
+    calculator = IncidentAnglesCalculator(
+        simtel_path=args_dict["simtel_path"],
+        db_config=db_config,
+        config_data=args_dict,
+        output_dir=output_dir,
+        label=base_label,
+    )
+    offsets = [float(v) for v in args_dict.get("off_axis_angles", [0.0])]
+
+    results_by_offset = calculator.run_for_offsets(offsets)
+    plot_incident_angles(
+        results_by_offset,
+        output_dir,
+        label_with_telescope,
+        debug_plots=args_dict.get("debug_plots", False),
+    )
+    total = sum(len(t) for t in results_by_offset.values())
+    logger.info(
+        f"Calculated incident angles for {len(results_by_offset)} offsets,\n"
```

Comment:
As discussed: photon statistics is not the important quantity. If you want to give a quick feedback, then better to print the mean of the distribution - the average incidence angle is important feedback.